### ISSUE TITLE
IncorrectObjectTypeException in RefWithCommitIterator: downgrade log level

### ISSUE THREAD

**Issue by zurk** [src-d/engine#354]
## Expected Behavior
See warning messages without 
```
WARN RefWithCommitIterator: incorrect object found
org.eclipse.jgit.errors.IncorrectObjectTypeException: Object d036093094b7056de592c29605d9210f57a145e0 is not a commit.
```
It hide spark progress bar and other usefull `WARN` messages, for example from bblfsh tool or java-related warnings.

## Current Behavior
Several identical warning messages in a second and no option to turn it off without useful logs loss.

## Possible Solution
1. Downgrade log level for such messages. As I understand it is normal that some objects are not commits, so nothing to warn about.
2. If it is not good to do it, good idea to have a flag or any kind of option that mute such messages.

## Context
I want to look through any suspicious warnings while engine process huge amount of data.

**Comment by bzz** [comment]
@carlosms with the great work you have already done on RefWithCommitIterator under #353 - would you mind taking a look into this one later on?

**Comment by carlosms** [comment]
Implemented in #359.

**Comment by carlosms** [comment]
Should this one be closed now that #359 is merged?
